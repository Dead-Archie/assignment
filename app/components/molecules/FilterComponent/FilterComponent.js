import React from 'react';
import PropTypes from 'prop-types';
import Router from 'next/router';
import { queryParamStr } from '../../../utils/utils';

// eslint-disable-next-line react/prefer-stateless-function
class FilterComponent extends React.Component {
  constructor(props) {
    super(props);
    this.queryString = [];
    this.state = {};
  }

  handleChangeCheckbox = event => {
    const { title, query } = this.props;
    const text = event.currentTarget.value;
    const isCheked = event.currentTarget.checked;

    if (isCheked && text) {
      query[title] = text;
    }
    // }
    const queryStr = queryParamStr(query);

    if (queryStr) {
      Router.push(`/?${queryStr}`);
    } else {
      Router.push(`/`);
    }
  };

  render() {
    const { data, title, query } = this.props;
    return (
      <div>
        <h4>{title}</h4>
        {data.map((item, index) => {
          return (
            <div className="filter-options">
              <input
                type="radio"
                id={`${title}_${index}`}
                value={item}
                onChange={event => this.handleChangeCheckbox(event)}
                checked={query[title] === item}
                name={title}
              />
              <label htmlFor={`${title}_${index}`}>{item}</label>
            </div>
          );
        })}
      </div>
    );
  }
}

FilterComponent.propTypes = {
  data: PropTypes.shape([]),
  title: PropTypes.string,
  query: PropTypes.shape({}),
};
FilterComponent.defaultProps = {
  data: [],
  title: '',
  query: {},
};

export default FilterComponent;
export { FilterComponent };
