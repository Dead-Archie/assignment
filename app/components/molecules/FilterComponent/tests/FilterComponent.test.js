import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { FilterComponent } from '../FilterComponent';

jest.mock('next/router', () => ({ push: () => {} }));

const sampleProps = {
  className: 'withStyles-sc-4wveuc-0 lgzevM',
  query: { gender: 'male', name: 'morty', species: 'Human' },
  title: 'origin',
  data: ['Earth (C-137)', 'unknown', 'Eric Stoltz Mask Eart'],
};

const setUp = props => {
  return shallow(<FilterComponent {...props} />);
};

describe('<FilterComponent />', () => {
  let FilterComponentPage;
  beforeEach(() => {
    FilterComponentPage = setUp(sampleProps);
  });

  test('should render correctly', () => {
    expect(FilterComponentPage).toMatchSnapshot();
  });

  test('cases for rendering filter options', () => {
    const wrapper = setUp(sampleProps);
    const filterOptions = wrapper.find('.filter-options');
    expect(filterOptions).toHaveLength(sampleProps.data.length);
  });
  test('cases for selecting check boxes', () => {
    const wrapper = setUp(sampleProps);
    const event = {
      currentTarget: {
        value: 'Earth (C-137)',
        checked: true,
      },
    };

    const checkBox = wrapper.find('input');
    console.log(checkBox);
    checkBox.map(item => item.simulate('change', event));
  });
});
