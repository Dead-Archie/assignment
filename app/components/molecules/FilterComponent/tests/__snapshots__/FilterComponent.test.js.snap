// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterComponent /> should render correctly 1`] = `
<div>
  <h4>
    origin
  </h4>
  <div
    className="filter-options"
  >
    <input
      checked={false}
      id="origin_0"
      name="origin"
      onChange={[Function]}
      type="radio"
      value="Earth (C-137)"
    />
    <label
      htmlFor="origin_0"
    >
      Earth (C-137)
    </label>
  </div>
  <div
    className="filter-options"
  >
    <input
      checked={false}
      id="origin_1"
      name="origin"
      onChange={[Function]}
      type="radio"
      value="unknown"
    />
    <label
      htmlFor="origin_1"
    >
      unknown
    </label>
  </div>
  <div
    className="filter-options"
  >
    <input
      checked={false}
      id="origin_2"
      name="origin"
      onChange={[Function]}
      type="radio"
      value="Eric Stoltz Mask Eart"
    />
    <label
      htmlFor="origin_2"
    >
      Eric Stoltz Mask Eart
    </label>
  </div>
</div>
`;
